src/onnx/parse_matmul.cpp

```diff
@@ -38,6 +38,100 @@ struct parse_matmul : op_parser<parse_matmul>
         return {{"MatMul", "dot"}, {"MatMulInteger", "quant_dot"}};
     }
 
+    static void broadcast_dimensions(const onnx_parser::node_info& info,
+                                     const std::vector<size_t>& s0_lens,
+                                     const std::vector<size_t>& s1_lens,
+                                     instruction_ref& a0,
+                                     instruction_ref& a1,
+                                     instruction_ref& ba0,
+                                     instruction_ref& ba1)
+    {
+        // try broadcasting if dimensions other than last two do not match
+        if(not std::equal(
+               s0_lens.rbegin() + 2, s0_lens.rend(), s1_lens.rbegin() + 2, s1_lens.rend()))
+        {
+            auto l0_it = s0_lens.begin() + s0_lens.size() - 2;
+            std::vector<std::size_t> l0_broadcasted_lens(s0_lens.begin(), l0_it);
+            auto l1_it = s1_lens.begin() + s1_lens.size() - 2;
+            std::vector<std::size_t> l1_broadcasted_lens(s1_lens.begin(), l1_it);
+            auto output_lens = compute_broadcasted_lens(l0_broadcasted_lens, l1_broadcasted_lens);
+            l0_broadcasted_lens = output_lens;
+            l0_broadcasted_lens.insert(l0_broadcasted_lens.end(), l0_it, s0_lens.end());
+            l1_broadcasted_lens = output_lens;
+            l1_broadcasted_lens.insert(l1_broadcasted_lens.end(), l1_it, s1_lens.end());
+            if(s0_lens != l0_broadcasted_lens)
+            {
+                ba0 = info.add_instruction(
+                    make_op("multibroadcast", {{"out_lens", l0_broadcasted_lens}}), a0);
+            }
+            if(s1_lens != l1_broadcasted_lens)
+            {
+                ba1 = info.add_instruction(
+                    make_op("multibroadcast", {{"out_lens", l1_broadcasted_lens}}), a1);
+            }
+        }
+    }
+
+    static void
+    add_int8_shift(const onnx_parser::node_info& info, instruction_ref& ba0, instruction_ref& ba1)
+    {
+        const auto ba0_type = ba0->get_shape().type();
+        const auto ba1_type = ba1->get_shape().type();
+
+        // determine if we need to add a shift to ensure items get shifted to the same int8
+        // type if we see uint8/int8 mismatches
+        if(ba0_type == ba1_type)
+            return;
+
+        instruction_ref unshifted_input;
+        instruction_ref* shifted_output = nullptr;
+        if(ba0_type == migraphx::shape::int8_type)
+        {
+            unshifted_input = ba1;
+            shifted_output  = &ba1;
+        }
+        if(ba1_type == migraphx::shape::int8_type)
```

Comment:
make it `else if` and add `else` clause that throws. Right now `MatMulInteger` is only supporting int8/uint8 but that may change in future to include FP8.

Reply:
You want to see a supported types lookup then? That should be on input for the quant_dot then as this should just check if types are the same but would pass if we have A,B inputs + zero points all be fp8s

Reply:
Added support for specific types and test coverage on its failure